bouncer/shared/setup_lp_account.ts

```diff
@@ -0,0 +1,96 @@
+import type { HexString } from '@polkadot/util/types';
+import { fundStateChainAccount } from '@chainflip/cli';
+import { Wallet, ethers } from 'ethers';
+import { Keyring } from '../polkadot/keyring';
+import { getNextEvmNonce } from './send_evm';
+import {
+  getContractAddress,
+  hexPubkeyToFlipAddress,
+  decodeFlipAddressForContract,
+  getEvmEndpoint,
+  getWhaleKey,
+  assetDecimals,
+  lpMutex,
+  handleSubstrateError,
+  amountToFineAmount,
+} from './utils';
+import { approveErc20 } from './approve_erc20';
+import { getChainflipApi, observeEvent } from './utils/substrate';
+
+export async function setupLpAccount(lpKey: string) {
```

Comment:
This whole function is *very* similar to fund_flip - we should be able to share a lot more code here

Reply:
Actually my bad here I don't know why I copy pasted the code of fund_flip instead of simply calling it :/
Anyway fixed the comments in fund_flip directly